cli/pcluster/config/cfn_param_types.py

```diff
@@ -1061,7 +1061,7 @@ def to_storage(self, storage_params=None):
                     param = param_type(self.key, self.label, param_key, param_definition, self.pcluster_config)
                     cfn_items.append(param.get_cfn_value())
 
-            if cfn_items[0] == "NONE":
+            if cfn_items[0] == "NONE" and not self.definition.get("dont_convert_to_NONE"):
```

Comment:
Let's remove this old approach.

Here we were checking if the first item was NONE, and if so we were setting all the items to NONE, because we wanted to change all the default values to NONE.

Example in the RAIDSection case, by looking at the mappings you can see that some params have a default value, so the RAIDOptions CFN param would be:
`<NONE,NONE,2,gp2,20,100,false,NONE>`
but we want to convert it to `<NONE,NONE,NONE,NONE,NONE,NONE,NONE,NONE>` so we were executing the following code, to reset all the values to NONE: 
```
if cfn_items[0] == "NONE"
     cfn_items = ["NONE"] * len(self.definition.get("params"))
``` 

I'd remove this approach. We can get rid of both the adaptation code before and the extra attribute `dont_convert_to_NONE`.
We can instead define a new `cfn_default` attribute.

We need to set `cfn_default` for all the options in the mappings that have a "default" value (e.g. int, boolean or string) but must be converted to NONE during the `to_storage` conversion.

I mean for example:
```python
            ("num_of_raid_volumes", {
                "default": 2,   # the param has a default value but we want to convert to NONE
                "cfn_default": "NONE",  # we can explicitly say that this must be converted to none when used for CFN
```

In this way we'll have a more generic solution that can work for all the old parameters that have a default (on which we can add `"cfn_default": "NONE",`) and for the new parameters for FSx HDD, on which we can define "cfn_default": "NONEDEFAULT",.


So we can remove the adapation code:
```
if cfn_items[0] == "NONE"
     cfn_items = ["NONE"] * len(self.definition.get("params"))
```
because by using the `definition.get("cfn_default", "NONE")` approach all the values will be the correct ones or NONE, without the need to convert all of them to NONE.

Reply:
Note, after using the cfn_default approach for al lthe parameter you can remove the `cfn_items = ["NONE"] * len(self.definition.get("params"))` from `EFSCfnSection` too.